### PR TITLE
Use index hint for homepage query

### PR DESCRIPTION
**Pull request by bladealslayer**
DB server doesn't seem to always pick the correct index, so we give it a hint.

### CODE REVIEW

File: app/services/listing_index_service/search/database_search_helper.rb [hunk 1]
```diff
@@ -16,11 +16,21 @@ def fetch_from_db(community_id:, search:, included_models:, includes:)
...
-    query = Listing
-            .where(where_opts)
-            .includes(included_models)
-            .order("listings.sort_date DESC")
-            .paginate(per_page: search[:per_page], page: search[:page])
+    query = if !search[:include_closed] && !search[:author_id]
+              # Add index hint for homepage query
+              Listing
+                .use_homepage_index
+                .where(where_opts)
```
Comment (aivils) [inline app/services/listing_index_service/search/database_search_helper.rb]: @bladealslayer 
scope is chainable

```
scope = Listing
scope = scope.use_homepage_index !search[:include_closed] && !search[:author_id]
scope = scope.currently_open unless search[:include_closed]
query = scope.where(where_opts)
                .includes(included_models)
                .order("listings.sort_date DESC")
                .paginate(per_page: search[:per_page], page: search[:page])
```

Comment (bladealslayer) [reply app/services/listing_index_service/search/database_search_helper.rb]: Indeed, thanks!

**Comment by bladealslayer** [comment]
@aivils does this look ok to you now?

**Comment by bladealslayer** [comment]
@aivils you you means something like:

```ruby
  HOMEPAGE_INDEX = "listings_homepage_query"
  scope :use_index, ->(index) { from("#{self.table_name} USE INDEX (#{index})") }
  scope :use_homepage_index, -> { use_index(HOMEPAGE_INDEX) }
```

**Pull request merged**